cargo-miri/bin.rs

```diff
@@ -1181,6 +1190,13 @@ fn main() {
             let binary = Path::new(arg);
             if binary.exists() {
                 assert!(!arg.starts_with("--")); // not a flag
+                if let Some(rustc) = std::env::var_os("RUSTC") {
+                    if rustc == arg {
+                        // In bootstrap we can't rely on rustc just being named rustc
```

Comment:
```suggestion
                        // This can only happen in bootstrap; we usually clear RUSTC precisely
                        // to avoid this situation.
```

Reply:
I'd prefer to just not clear the rustc env var at all and always use this logic

Reply:
Keeping the RUSTC env var is IMO semantically wrong -- we should not be invoking rustc, only Miri.

Also this could cause a lot of trouble if RUSTC is some relative path that also happens to be used for a binary during the build. Then we'll end up confusing our phases here.

Reply:
> Also this could cause a lot of trouble if RUSTC is some relative path that also happens to be used for a binary during the build. Then we'll end up confusing our phases here.

why would relative paths be relevant here? (that said, I am working on getting this to work via miri, but it looks nontrivial, so if I'm not done by tomorrow noon, I won't get to it again until monday.)

Reply:
> why would relative paths be relevant here?

We are set as the TARGET_RUNNER by cargo. So what if there is a binary called "foobar" in that project but we also have RUSTC=foobar (pointing to a different foobar in the PATH)? Then everything will be very confusing.